hangman.rb

```diff
@@ -0,0 +1,68 @@
+
+$random_words = ['ruby', 'hangman', 'flutter', 'developer', 'omkar', 'github', 'josh', 'assignment']
+
+def choose_word
+  $random_words.sample
+end
+
+def display_word(word,correct_guesses)
+  word.chars.map { |char| correct_guesses.include?(char) ? char : '_' }.join(' ')
+end
+ 
+
+def check_game_result(word,remaining_count)
+  if remaining_count == 0
+    puts "You guessed the word '#{word}' correctly"
+    return
+  end
+  
+  puts "Better luck next time, the word was '#{word}'"
+end
```

Comment:
Dont print here, print where you are calling this from, this will make this method concise

Reply:
You can do the same with the process_guess method, return the remaining_count and message, and print it inside the play_game methods loop.